### PR TITLE
Fixes for #89 and #91

### PR DESCRIPTION
**Pull request by jorgemoralespou**
Adding the ability to specify MAVEN_SETTINGS, MAVEN_MIRROR_URL and a  default settings.xml and copy .ear, .war, .rar and .jar

* This PR allows to deploy .ear .war .jar and .rar
* Allows to provide own settings.xml through MAVEN_SETTINGS
* Allows to provide MAVEN_MIRROR_URL in default settings.xml in the images

This PR is same as #92 but squashed

### CODE REVIEW

File: 10.0/s2i/bin/assemble [hunk 1]
```diff
@@ -150,6 +150,64 @@ function ismgmtup() {
...
+         <active>true</active>\
+         <protocol>http</protocol>\
+         <host>$HTTP_PROXY_HOST</host>\
+         <port>$HTTP_PROXY_PORT</port>"
+    if [ -n "$HTTP_PROXY_USERNAME" -a -n "$HTTP_PROXY_PASSWORD" ]; then
+      xml="$xml\
+         <username>$HTTP_PROXY_USERNAME</username>\
+         <password>$HTTP_PROXY_PASSWORD</password>"
+    fi
+    if [ -n "$HTTP_PROXY_NONPROXYHOSTS" ]; then
```
Comment (bparees) [inline 10.0/s2i/bin/assemble]: isn't NO_PROXY the standard name for this env var?

Comment (bparees) [inline 10.0/s2i/bin/assemble]: i guess since it matches what EAP uses, we can live with it, but NO_PROXY probably would have been a better env var to use for this.

**Comment by bparees** [comment]
lgtm, will merge if tests pass.

File: hack/build.sh [hunk 2]
```diff
@@ -88,7 +88,7 @@ for dir in ${dirs}; do
     docker_build_with_version Dockerfile
 #  fi
 
-  if [[ -v TEST_MODE ]]; then
+  if [ -v TEST_MODE ]; then
```
Comment (bparees) [inline hack/build.sh]: why?

File: hack/build.sh [hunk 3]
```diff
@@ -75,7 +75,7 @@ for dir in ${dirs}; do
 
   IMAGE_NAME="${NAMESPACE}${BASE_IMAGE_NAME}-${dir//./}-${OS}"
 
-  if [[ -v TEST_MODE ]]; then
+  if [ -v TEST_MODE ]; then
```
Comment (bparees) [inline hack/build.sh]: why?

File: 10.0/s2i/bin/assemble (hunk 1 above)
Comment (jorgemoralespou) [inline 10.0/s2i/bin/assemble]: I would agree, but EAP uses that:

HTTP_PROXY_NONPROXYHOSTS: If supplied, a configured HTTP proxy will ignore these hosts. (some.example.org|*.example.net)

File: hack/build.sh (hunk 2 above)
Comment (jorgemoralespou) [inline hack/build.sh]: It failed for me saying that expected unary operator.

File: hack/build.sh (hunk 3 above)
Comment (jorgemoralespou) [inline hack/build.sh]: Same reason

File: hack/build.sh (hunk 2 above)
Comment (bparees) [inline hack/build.sh]: were you trying to run the script directly?  TEST_MODE needs to be set.

Comment (jorgemoralespou) [inline hack/build.sh]: Do you want me to revert this change?

**Comment by jorgemoralespou** [comment]
Fixed in #94  or #95
